tests/validations/mta/mtaValidations.spec.ts

```diff
@@ -94,28 +103,72 @@ describe("mtaValidations", () => {
       await testUpdateMtaDiagnostics([]);
     });
 
+    it("returns no diagnostics when mta.yaml does not exist and clear the current diagnostics", async () => {
+      const expectedDiagnostics: [Uri, Diagnostic[] | undefined][] = [
+        [
+          // when clearing the diagnostic, the diagnostic remains in the cache with "undefined"
+          mtaYamlUri,
+          undefined,
+        ],
+      ];
+      const currentCollectionEntries: [Uri, Diagnostic[] | undefined][] = [
+        [
+          mtaYamlUri,
+          [
+            {
+              source: "MTA",
+              message: "diagnostic message",
+              range: new Range(0, 0, 0, 0),
+              severity: 0,
+            },
+          ],
+        ],
+      ];
+      await testUpdateMtaDiagnostics(
+        expectedDiagnostics,
+        currentCollectionEntries
+      );
+    });
+
     it("returns no diagnostics when mta.yaml and dev.mtaext have no errors", async () => {
-      sinon.stub(fsExtra, "pathExists").resolves(true);
+      stubPathExists(
+        { path: mtaYamlPath, resolvesTo: true },
+        { path: devMtaExtPath, resolvesTo: true }
+      );
       sinon.stub(Mta.prototype, "validate").resolves({});
 
       await testUpdateMtaDiagnostics([]);
     });
 
     it("returns diagnostics when mta.yaml and dev.mtaext have errors", async () => {
-      const expectedEntries = prepDiagnostics();
+      const [
+        expectedEntries,
+        getValidationSpy,
+      ] = prepValidationAndDiagnostics();
       await testUpdateMtaDiagnostics(expectedEntries);
+
+      expect(getValidationSpy.callCount).to.equal(1);
+      expect(getValidationSpy.withArgs(modulePath, [devMtaExtPath]).called).to
+        .be.true;
     });
 
-    async function testUpdateMtaDiagnostics(
-      expectedCollectionEntries: [Uri, Diagnostic[] | undefined][]
-    ) {
-      const uri = {
-        fsPath: mtaYamlPath,
-      } as Uri;
+    it("returns diagnostics when mta.yaml have errors and dev.mtaext does not exist", async () => {
+      const [expectedEntries, getValidationSpy] = prepValidationAndDiagnostics(
+        false
+      );
+      await testUpdateMtaDiagnostics(expectedEntries);
 
-      let collectionEntries:
+      expect(getValidationSpy.callCount).to.equal(1);
+      expect(getValidationSpy.withArgs(modulePath, undefined).called).to.be
```

Comment:
Please use `expect(getValidationSpy.firstCall.args).to.deep.equal([modulePath, undefined]);` to have a good error message in case of failure